tests/flytekit/unit/experimental/test_eager_workflows.py

```diff
@@ -275,3 +281,26 @@ async def eager_wf_flyte_directory() -> str:
 
     result = asyncio.run(eager_wf_flyte_directory())
     assert result == "some data"
+
+
+@mock.patch("flytekit.core.utils.load_proto_from_file")
+@mock.patch("flytekit.core.data_persistence.FileAccessProvider.get_data")
+@mock.patch("flytekit.core.data_persistence.FileAccessProvider.put_data")
+@mock.patch("flytekit.core.utils.write_proto_to_file")
+def test_eager_workflow_dispatch(mock_write_to_file, mock_put_data, mock_get_data, mock_load_proto, event_loop):
+    """Test that event loop is preserved after executing eager workflow via dispatch."""
+
+    @eager
+    async def eager_wf():
+        await asyncio.sleep(0.1)
+        return
+
+    ctx = context_manager.FlyteContext.current_context()
+    with context_manager.FlyteContextManager.with_context(
+        ctx.with_execution_state(
+            ctx.execution_state.with_params(mode=context_manager.ExecutionState.Mode.TASK_EXECUTION)
+        )
+    ) as ctx:
+        _dispatch_execute(ctx, lambda: eager_wf, "inputs path", "outputs prefix")
+        loop_after_execute = asyncio.get_event_loop_policy().get_event_loop()
+        assert event_loop == loop_after_execute
```

Comment:
assuming the fix goes in on the union side, i still feel like this assert should fail.

Reply:
posted some thoughts on the other pr, which basically wraps the whole _dispatch_execute call in an event loop so one is there if necessary.